weave-js/src/components/Panel2/PanelString.tsx

```diff
@@ -53,6 +53,27 @@ function isURL(text: string): boolean {
   }
 }
 
+/**
+ * Determines if a given text contains any right-to-left (RTL) script characters.
+ *
+ * RTL script characters are defined as any character within the Unicode ranges:
+ * - U+0590 to U+05FF
+ * - U+0600 to U+06FF
+ * - U+0750 to U+077F
+ * - U+08A0 to U+08FF
+ * - U+FB50 to U+FDFF
+ * - U+FE70 to U+FEFF
+ * Matching arabic, hebrew, and other related scripts.
+ *
+ * @param text - The text to be checked for RTL characters.
+ * @returns `true` if the text contains RTL characters, otherwise `false`.
+ */
+const isRTL = (text: string): boolean => {
+  const rtlChars =
```

Comment:
you could technically compile this first. If we are running this against all text in every panel we might want to take that micro perf gain.